source/GameInterface/Services/MobileParties/Patches/MobilePartyFieldPatches.cs

```diff
@@ -0,0 +1,892 @@
+public class MobilePartyFieldPatches
+{
+    private static readonly ILogger Logger = LogManager.GetLogger<MobilePartyFieldPatches>();
+
+    private static IEnumerable<MethodBase> TargetMethods()
+    {
+        return AccessTools.GetDeclaredMethods(typeof(MobileParty));
+    }
+    
+    [HarmonyTranspiler]
+    private static IEnumerable<CodeInstruction> AttachedToTranspiler(IEnumerable<CodeInstruction> instructions)
+    {
+        var field = AccessTools.Field(typeof(MobileParty), nameof(MobileParty._attachedTo));
+        var fieldIntercept = AccessTools.Method(typeof(MobilePartyFieldPatches), nameof(AttachedToIntercept));
+
+        foreach (var instruction in instructions)
+        {
+            if (instruction.StoresField(field))
+            {
+                yield return new CodeInstruction(OpCodes.Call, fieldIntercept);
+            }
+            else
+            {
+                yield return instruction;
+            }
+        }
+    }
+
+    public static void AttachedToIntercept(MobileParty instance, MobileParty newAttachedTo)
+    {
+        if (CallOriginalPolicy.IsOriginalAllowed())
+        {
+            instance._attachedTo = newAttachedTo;
+            return;
+        }
+        if (ModInformation.IsClient)
+        {
+            Logger.Error("Client added unmanaged item: {callstack}", Environment.StackTrace);
+            instance._attachedTo = newAttachedTo;
+            return;
+        }
+
+        MessageBroker.Instance.Publish(instance, new AttachedToChanged(newAttachedTo.StringId, instance.StringId));
+
+        instance._attachedTo = newAttachedTo;
+    }
+
+    [HarmonyTranspiler]
+    private static IEnumerable<CodeInstruction> HasUnpaidWagesTranspiler(IEnumerable<CodeInstruction> instructions)
+    {
+        var field = AccessTools.Field(typeof(MobileParty), nameof(MobileParty.HasUnpaidWages));
+        var fieldIntercept = AccessTools.Method(typeof(MobilePartyFieldPatches), nameof(HasUnpaidWagesIntercept));
+
+        foreach (var instruction in instructions)
+        {
+            if (instruction.StoresField(field))
+            {
+                yield return new CodeInstruction(OpCodes.Call, fieldIntercept);
+            }
+            else
+            {
+                yield return instruction;
+            }
+        }
+    }
+
+    public static void HasUnpaidWagesIntercept(MobileParty instance, float newHasUnpaidWages)
+    {
+        if (CallOriginalPolicy.IsOriginalAllowed())
+        {
+            instance.HasUnpaidWages = newHasUnpaidWages;
+            return;
+        }
+        if (ModInformation.IsClient)
+        {
+            Logger.Error("Client added unmanaged item: {callstack}", Environment.StackTrace);
+            instance.HasUnpaidWages = newHasUnpaidWages;
+            return;
+        }
+
+        MessageBroker.Instance.Publish(instance, new HasUnpaidWagesChanged(newHasUnpaidWages, instance.StringId));
+
+        instance.HasUnpaidWages = newHasUnpaidWages;
+    }
+    /*
```

Comment:
I did that because of the poor performance. After the fix with the ProtoMember attributes it is now better again, but still not optimal, probably due to the exceptions that occur now.
Do I need to do anything else here with the AllowedThread? Maybe call CalculateSpeed on the client side with it?

Reply:
Removed as discussed